src/lib/sql/sql_translator.cpp

```diff
@@ -29,92 +48,107 @@
+    {hsql::kOpAsterisk, ArithmeticOperator::Multiplication}, {hsql::kOpSlash, ArithmeticOperator::Division},
+    {hsql::kOpPercentage, ArithmeticOperator::Modulo},
+};
+
+const std::unordered_map<hsql::OperatorType, LogicalOperator> hsql_logical_operators = {
+    {hsql::kOpAnd, LogicalOperator::And}, {hsql::kOpOr, LogicalOperator::Or}};
+
+const std::unordered_map<hsql::OperatorType, PredicateCondition> hsql_predicate_condition = {
+    {hsql::kOpBetween, PredicateCondition::Between},
+    {hsql::kOpEquals, PredicateCondition::Equals},
+    {hsql::kOpNotEquals, PredicateCondition::NotEquals},
+    {hsql::kOpLess, PredicateCondition::LessThan},
+    {hsql::kOpLessEq, PredicateCondition::LessThanEquals},
+    {hsql::kOpGreater, PredicateCondition::GreaterThan},
+    {hsql::kOpGreaterEq, PredicateCondition::GreaterThanEquals},
+    {hsql::kOpLike, PredicateCondition::Like},
+    {hsql::kOpNotLike, PredicateCondition::NotLike},
+    {hsql::kOpIsNull, PredicateCondition::IsNull}};
+
+const std::unordered_map<hsql::DatetimeField, DatetimeComponent> hsql_datetime_field = {
+    {hsql::kDatetimeYear, DatetimeComponent::Year},     {hsql::kDatetimeMonth, DatetimeComponent::Month},
+    {hsql::kDatetimeDay, DatetimeComponent::Day},       {hsql::kDatetimeHour, DatetimeComponent::Hour},
+    {hsql::kDatetimeMinute, DatetimeComponent::Minute}, {hsql::kDatetimeSecond, DatetimeComponent::Second},
+};
+
+JoinMode translate_join_mode(const hsql::JoinType join_type) {
+  static const std::unordered_map<const hsql::JoinType, const JoinMode> join_type_to_mode = {
+      {hsql::kJoinInner, JoinMode::Inner}, {hsql::kJoinFull, JoinMode::Outer},  {hsql::kJoinLeft, JoinMode::Left},
+      {hsql::kJoinRight, JoinMode::Right}, {hsql::kJoinCross, JoinMode::Cross},
+  };
 
-PredicateCondition translate_operator_type_to_predicate_condition(const hsql::OperatorType operator_type) {
-  static const std::unordered_map<const hsql::OperatorType, const PredicateCondition> operator_to_predicate_condition =
-      {{hsql::kOpEquals, PredicateCondition::Equals},
-       {hsql::kOpNotEquals, PredicateCondition::NotEquals},
-       {hsql::kOpGreater, PredicateCondition::GreaterThan},
-       {hsql::kOpGreaterEq, PredicateCondition::GreaterThanEquals},
-       {hsql::kOpLess, PredicateCondition::LessThan},
-       {hsql::kOpLessEq, PredicateCondition::LessThanEquals},
-       {hsql::kOpBetween, PredicateCondition::Between},
-       {hsql::kOpLike, PredicateCondition::Like},
-       {hsql::kOpNotLike, PredicateCondition::NotLike},
-       {hsql::kOpIsNull, PredicateCondition::IsNull},
-       {hsql::kOpIn, PredicateCondition::In}};
-
-  auto it = operator_to_predicate_condition.find(operator_type);
-  DebugAssert(it != operator_to_predicate_condition.end(), "Filter expression clause operator is not yet supported.");
+  auto it = join_type_to_mode.find(join_type);
+  Assert(it != join_type_to_mode.end(), "Unknown to handle join type.");
```

Comment:
Typo: Unable and also AssertInput?!

Reply:
Fixed message, but for me, this is not an AssertInput as this will only break if the SQLparser delivers an invalid AST, not when the user makes a genuine mistake

Reply:
Agreed.